api/turing/cluster/servicebuilder/service_builder.go

```diff
@@ -305,6 +331,56 @@ func (sb *clusterSvcBuilder) NewSecret(
 	}
 }
 
+func (sb *clusterSvcBuilder) getMinReplicaOrDefault(resourceRequest *models.ResourceRequest) (*int, error) {
+	if resourceRequest == nil {
+		if sb.DefaultAutoscalingPolicyMinReplica == nil {
+			return nil, errors.New("API does not have min replica for default autoscaling policy set")
+		}
+		return sb.DefaultAutoscalingPolicyMinReplica, nil
+	}
+	return &resourceRequest.MinReplica, nil
+}
+
+func (sb *clusterSvcBuilder) getMaxReplicaOrDefault(resourceRequest *models.ResourceRequest) (*int, error) {
+	if resourceRequest == nil {
+		if sb.DefaultAutoscalingPolicyMaxReplica == nil {
+			return nil, errors.New("API does not have max replica for default autoscaling policy set")
+		}
+		return sb.DefaultAutoscalingPolicyMaxReplica, nil
+	}
+	return &resourceRequest.MaxReplica, nil
+}
+
+func (sb *clusterSvcBuilder) getCPURequestOrDefault(resourceRequest *models.ResourceRequest) resource.Quantity {
+	if resourceRequest == nil {
+		return sb.MaxCPU
+	}
+	return resourceRequest.CPURequest
+}
+
+func (sb *clusterSvcBuilder) getMemoryRequestOrDefault(resourceRequest *models.ResourceRequest) resource.Quantity {
+	if resourceRequest == nil {
+		return sb.MaxMemory
+	}
+	return resourceRequest.MemoryRequest
+}
+
+func (sb *clusterSvcBuilder) getAutoscalingMetricOrDefault(autoscalingPolicy models.AutoscalingPolicy) string {
+	// default autoscaling policy is used; hence use RPS as the default autoscaling policy metric
+	if autoscalingPolicy.PayloadSize != nil {
+		return string(models.AutoscalingMetricRPS)
+	}
+	return string(*autoscalingPolicy.Metric)
+}
+
+func (sb *clusterSvcBuilder) getAutoscalingTargetOrDefault(autoscalingPolicy models.AutoscalingPolicy) string {
+	// TODO: default autoscaling policy is used; hence calculate target RPS from payload size
+	if autoscalingPolicy.PayloadSize != nil {
+		return "500"
+	}
+	return *autoscalingPolicy.Target
+}
```

Comment:
This autoscaling target has temporarily been set as 500 rps when the default autoscaling policy is selected. A future PR will refine this method to set the initial target depending on the initial payload size.